plugins/scaffolder-backend-module-gitlab/src/actions/createGitlabProjectAccessTokenAction.ts

```diff
@@ -35,46 +37,103 @@ export const createGitlabProjectAccessTokenAction = (options: {
     id: 'gitlab:projectAccessToken:create',
     examples,
     schema: {
-      input: commonGitlabConfig.merge(
-        z.object({
-          projectId: z.union([z.number(), z.string()], {
-            description: 'Project ID',
-          }),
-          name: z.string({ description: 'Deploy Token Name' }).optional(),
-          accessLevel: z
-            .number({ description: 'Access Level of the Token' })
-            .optional(),
-          scopes: z.array(z.string(), { description: 'Scopes' }).optional(),
+      input: z.object({
+        projectId: z.number({
+          description: 'Project ID/Name(slug) of the Gitlab Project',
         }),
-      ),
+        name: z.string({ description: 'Name of Access Key' }).optional(),
+        repoUrl: z.string({ description: 'URL to gitlab instance' }),
+        accessLevel: z
+          .number({
+            description:
+              'Access Level of the Token, 10 (Guest), 20 (Reporter), 30 (Developer), 40 (Maintainer), and 50 (Owner)',
+          })
+          .optional(),
+        scopes: z
+          .array(
+            z.enum(
+              [
+                'api',
+                'read_api',
+                'create_runner',
+                'read_registry',
+                'write_registry',
+                'read_repository',
+                'write_repository',
+              ],
+              {
+                description: 'Scopes for a project access token',
+              },
+            ),
+          )
```

Comment:
Appreciate the more descriptive typing, but wondering if there already exists a Enum for this in `gitbeaker`, such that it won't break in case new scopes are added in the future :)

Reply:
Hi @tudi2d i have tested a Implemantation with the Scopes Type of the gitbeaker Package i zod.

const gitbeakerAccessTokenScopesType: z.ZodType<AccessTokenScopes[]> = z.any();
.
.
.
scopes: gitbeakerAccessTokenScopesType,

It works for allowed Scopes, but didn't validate the Input. 

Have you an Idea how to implement?

Thx

Thomas

Reply:
Not exactly sure, but I would expect that setting it to any would lead to no validation. What I meant with the prior comment was to check if there are existing types in `gitbeaker` you can reuse here instead of creating your own. 

I guess with `AccessTokenScopes` you would mean this types here, right: https://github.com/jdalrymple/gitbeaker/blob/main/packages/core/src/templates/ResourceAccessTokens.ts#L13-L20? You could use it instead of your custom enum.

Reply:
Ho @tudi2d,
what implementation we should do? The one with the static validation in zod or the one without validation in zod? 
Thanks Thomas